### PR TITLE
Cleanup handler validation

### PR DESCRIPTION
**Pull request by jcorbin**
r @kriskowal @Raynos

### CODE REVIEW

File: node/v2/handler.js [hunk 1]
```diff
@@ -183,14 +183,21 @@ TChannelV2Handler.prototype.handleInitResponse = function handleInitResponse(res
...
 
     var req = self.buildInRequest(reqFrame);
 
-    if (self.incomingRequestInvalid(reqFrame, req)) {
-        return null;
+    var err = self.checkCallReqFrame(reqFrame);
+    if (err) {
+        req.res = self.buildOutResponse(req);
+        self.sendErrorFrame(
+            req.res, 'ProtocolError', err.message
```
Comment (Raynos) [inline node/v2/handler.js]: We can switch here on the type of `err` and send an appropriate error type.

We can use `Errors.classify(err) || 'ProtocolError'`

Let's verify that the arg1 over length error is properly classified.

**Comment by Raynos** [comment]
Let's avoid the BadRequest regression; otherwise lgtm.

File: node/v2/handler.js (hunk 1 above)
Comment (jcorbin) [inline node/v2/handler.js]: Will do, good call; it's something I handle better later by simply emitting the err on req.errorEvent, which gets hooked up to such classification.  However I didn't want to add all of that in this PR.

Confirmed classification of arg1 over length, as of 4b45dec errors.js:578 has it as a `BadRequest`.

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**